src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -334,6 +310,165 @@ public CaseDTO getCaseByCaseReference(final long caseRef, CaseQueryRequestDTO re
     return caseServiceResponse;
   }
 
+  private void validateCompatibleEstabAndCaseType(CaseType caseType, EstabType estabType)
+      throws CTPException {
+    Optional<AddressType> addrType = estabType.getAddressType();
+    if (addrType.isPresent() && (caseType != CaseType.valueOf(addrType.get().name()))) {
+      log.with("caseType", caseType)
+          .with("estabType", estabType)
+          .info("Mismatching caseType and estabType");
+      String msg =
```

Comment:
I'd prob use a StringBuilder or such like, but it's up to you -

Reply:
Well that code was stolen and reused from somewhere else, so I can't claim ownership.  However as a rule, if we are in the exception path, the performance benefit is insignificant, so I wouldn't normally shy away from string concatenation which is way faster than the early days of java anyway.